### PR TITLE
Validation and generation of requst

### PR DESCRIPTION
**Pull request by OneWebPro**

### CODE REVIEW

File: src/main/scala/opentrack/jpaf/app/api/route/JourneyPlanRoute.scala [hunk 1]
```diff
@@ -19,23 +20,17 @@ trait JourneyPlanRoute {
 
   // /plan/:origin/:destination/:outwardDate/:returnDate/:numAdults/:numChildren/:railcards
   // e.g. http://localhost:8080/plan/TBW/HGS/2017-01-24/2017-01-24/1/0/YNG,DIS
+  // http://127.0.0.1:8080/plan?origin=TBW&destination=HGS&departureDate=2017-01-24&returnDate=2017-01-24&adults=1&children=0&railcards[]=YNG&railcards[]=DIS
   val journeyPlanRoutes: Route =
-    path("plan" / Segment / Segment / Segment / Segment / IntNumber / IntNumber/ Segment) { (origin, destination, outwardDate, returnDate, numAdults, numChildren, railcards) =>
+  path("plan") {
+    parameterSeq { params =>
```
Comment (linusnorton) [inline src/main/scala/opentrack/jpaf/app/api/route/JourneyPlanRoute.scala]: I think params is the better way to go

File: src/main/scala/opentrack/jpaf/app/api/validation/PlanJourneyRequest.scala [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  def apply(params: Params): Validation[JourneyPlanRequest] = {
+    val origin = StationField(params, "origin")
+    val destination = StationField(params, "destination")
+    val departureDate = DateField(params, "departureDate")
+    val returnDateOption = OptionalField(params, "returnDate", DateField)
+    val adults = IntegerField(params, "adults")
+    val children = IntegerField(params, "children")
+    val railcards = RailcardField(params, "railcards")
+
+    origin |@| destination |@| departureDate |@| returnDateOption |@| adults |@| children |@| railcards
```
Comment (linusnorton) [inline src/main/scala/opentrack/jpaf/app/api/validation/PlanJourneyRequest.scala]: I hate you.

File: src/main/scala/opentrack/jpaf/lib/journey/planner/JourneyPlanner.scala [hunk 3]
```diff
@@ -28,6 +28,19 @@ class JourneyPlanner(journeyService: JourneyService, faresService: FaresService)
     * @param request
     * @return
     */
+  def planJourney(request: JourneyPlanRequest): Future[JourneyPlanResults] =
+    request.returnDate match {
+      case Some(_) => this.planJourney(request.toReturn)
```
Comment (linusnorton) [inline src/main/scala/opentrack/jpaf/lib/journey/planner/JourneyPlanner.scala]: I can probably re-write some stuff to get rid of the two classes

Comment (OneWebPro) [inline src/main/scala/opentrack/jpaf/lib/journey/planner/JourneyPlanner.scala]: You can :) But maybe you had reason to have both types - its why I left it like that.

**Pull request merged**